src/PrestaShopBundle/Bridge/AdminController/FrameworkBridgeControllerListTrait.php

```diff
@@ -28,91 +28,64 @@
 
 namespace PrestaShopBundle\Bridge\AdminController;
 
-use PrestaShopBundle\Bridge\AdminController\Action\ActionInterface;
-use PrestaShopBundle\Bridge\Exception\NotAllowedActionTypeForListException;
-use PrestaShopBundle\Bridge\Helper\Listing\Action\ListBulkAction;
-use PrestaShopBundle\Bridge\Helper\Listing\Action\ListHeaderToolbarAction;
-use PrestaShopBundle\Bridge\Helper\Listing\Action\ListRowAction;
-use PrestaShopBundle\Bridge\Helper\Listing\Field\FieldInterface;
-use PrestaShopBundle\Bridge\Helper\Listing\FiltersProcessor;
 use PrestaShopBundle\Bridge\Helper\Listing\HelperListConfiguration;
+use Symfony\Component\HttpFoundation\Request;
 
 /**
  * Contains the principal methods you need to horizontally migrate a controller which has a list.
  */
 trait FrameworkBridgeControllerListTrait
 {
+    /**
+     * @param string $identifierKey
+     * @param string $positionIdentifierKey
+     * @param string $defaultOrderBy
+     * @param string|null $postSubmitRoute
+     * @param bool $autoJoinLangTable
+     * @param bool $deleted
+     * @param bool $explicitSelect
+     * @param bool $useFoundRows
+     * @param string|null $listId
+     *
+     * @return HelperListConfiguration
+     */
     protected function buildListConfiguration(
         string $identifierKey,
         string $positionIdentifierKey,
         string $defaultOrderBy,
+        ?string $postSubmitRoute = null,
         bool $autoJoinLangTable = true,
         bool $deleted = false,
         bool $explicitSelect = false,
-        bool $useFoundRows = true
+        bool $useFoundRows = true,
+        ?string $listId = null
     ): HelperListConfiguration {
         $controllerConfiguration = $this->getControllerConfiguration();
 
         return $this->get('prestashop.bridge.helper.listing.helper_list_configuration_factory')->create(
             $controllerConfiguration,
             $identifierKey,
+            $postSubmitRoute ?: $this->get('request_stack')->getCurrentRequest()->attributes->get('_route'),
```

Comment:
I don't think this "magic" part should be hidden in the trait, especially because I'm not confident about relying on the `request_stack` service, I've had a few issues in the past when I had to use it and now I'm really cautious

It's safer to keep a simple and explicit interface in the trait, if the controller wants to rely on the request routing it can use it internally since it already has the `$request` parameter available It can also provide it hard-coded but it's the controller's choice to decide how it gets back the route not the trait Too much dark magic can be dangerous 😅